### PR TITLE
Move s3_daily_storage and s3_request metricsets to use cloudwatch input

### PR DESCRIPTION
**Pull request by kaiyan-sheng**
## What does this PR do?

This PR is to move s3_daily_storage and s3_request metricsets to use cloudwatch input. In order to make it not a breaking change, `rename` processor is used in both `manifest.yml` files.

## Why is it important?

This way we only need to maintain code in cloudwatch input instead of maintaining code in each metricset.

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] ~~I have commented my code, particularly in hard-to-understand areas~~
- [ ] ~~I have made corresponding changes to the documentation~~
- [ ] ~~I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

File: x-pack/metricbeat/module/aws/s3_daily_storage/_meta/data.json [hunk 1]
```diff
@@ -1,27 +1,28 @@
...
+            "StorageType": "AllStorageTypes"
         },
-        "s3_daily_storage": {
-            "bucket": {
-                "size": {
-                    "bytes": 207372
+        "s3": {
+            "metrics": {
+                "NumberOfObjects": {
+                    "avg": 57828
```
Comment (jsoriano) [inline x-pack/metricbeat/module/aws/s3_daily_storage/_meta/data.json:14]: Is rename not being applied here?

File: x-pack/metricbeat/module/aws/s3_request/_meta/data.json [hunk 2]
```diff
@@ -1,40 +1,46 @@
...
+                },
+                "FirstByteLatency": {
+                    "avg": 21.791666666666668
+                },
+                "PutRequests": {
+                    "avg": 1
+                },
+                "TotalRequestLatency": {
+                    "avg": 35.916666666666664
+                }
```
Comment (jsoriano) [inline x-pack/metricbeat/module/aws/s3_request/_meta/data.json:33]: Also here, are renames not being applied?

**Review by jsoriano — COMMENTED** [review]
Are `data.json` files updated?

File: x-pack/metricbeat/module/aws/s3_daily_storage/_meta/data.json (hunk 1 above)
Comment (kaiyan-sheng) [reply x-pack/metricbeat/module/aws/s3_daily_storage/_meta/data.json:14]: Yeah it's applied when I run metricbeat for testing but not applied when I tried to create the event for data.json. Is there any way I can make the rename processor work with `TestData` function?

File: x-pack/metricbeat/module/aws/s3_request/s3_request_integration_test.go [hunk 3]
```diff
@@ -10,54 +10,13 @@ package s3_request
...
-		mtest.CheckEventField("select_returned.bytes", "float", event, t)
-		mtest.CheckEventField("requests.list", "int", event, t)
-		mtest.CheckEventField("downloaded.bytes", "float", event, t)
-		mtest.CheckEventField("uploaded.bytes", "float", event, t)
-		mtest.CheckEventField("errors.4xx", "int", event, t)
-		mtest.CheckEventField("errors.5xx", "int", event, t)
-		mtest.CheckEventField("latency.first_byte.ms", "float", event, t)
-		mtest.CheckEventField("latency.total_request.ms", "float", event, t)
-	}
-}
```
[the remark this answers is not in the document]
Comment (kaiyan-sheng) [reply x-pack/metricbeat/module/aws/s3_request/s3_request_integration_test.go:54]: Ahh yes I copied pasted a test from other file 😂 Let me add these back in! Thanks!

File: x-pack/metricbeat/module/aws/s3_daily_storage/_meta/data.json (hunk 1 above)
Comment (jsoriano) [reply x-pack/metricbeat/module/aws/s3_daily_storage/_meta/data.json:14]: Oh ok, maybe processors are not initialized on testing, if that is the case we should fix that, but not on this PR. Could you please open an issue about that?

But interestingly in the iis module there is a rename that seems to be applied :thinking: 
https://github.com/elastic/beats/blob/3e5a167809210e9999732d5e50c7e895e10fc389/x-pack/metricbeat/module/iis/website/manifest.yml#L36-L40
https://github.com/elastic/beats/blob/3e5a167809210e9999732d5e50c7e895e10fc389/x-pack/metricbeat/module/iis/website/_meta/data.json#L27

**Review by jsoriano — APPROVED** [review]
Looks good, but we should check why processors are not being executed on `TestData`.

File: x-pack/metricbeat/module/aws/s3_daily_storage/_meta/data.json (hunk 1 above)
Comment (kaiyan-sheng) [reply x-pack/metricbeat/module/aws/s3_daily_storage/_meta/data.json:14]: Interesting. @narph I saw you generated the data.json file for iis website. Does the field `iis.website.name` generated automatically by `TestData` function? For my case, `TestData` function ignores the rename processor.

Comment (narph) [reply x-pack/metricbeat/module/aws/s3_daily_storage/_meta/data.json:14]: I am setting the `iis.website.name` using the rename processor inside the manifest.yml file

**Pull request merged**